pkg/security/keychain.go

```diff
@@ -58,14 +67,176 @@ func SecKeyUpdate(connectionID string, username string, password string) *SecErr
 }
 
 // SecKeyGetSecret : retrieve secret / credentials from the keyring
-func SecKeyGetSecret(connectionID string, username string) (string, *SecError) {
-
+func SecKeyGetSecret(connectionID, username string) (string, *SecError) {
 	conID := strings.TrimSpace(strings.ToLower(connectionID))
 	uName := strings.TrimSpace(strings.ToLower(username))
+	secret, err := GetSecretFromKeyring(conID, uName)
+	if err != nil {
+		return "", err
+	}
+	return secret, nil
+}
+
+// StoreSecretInKeyring stores the secret in either the system keyring or our insecure keyring.
+func StoreSecretInKeyring(connectionID, uName, pass string) *SecError {
+	service := connectionIDToService(connectionID)
+	if globals.UseInsecureKeyring {
+		_, statErr := os.Stat(GetPathToInsecureKeyring())
+		if os.IsNotExist(statErr) {
+			os.MkdirAll(insecureKeyringDir, 0600)
+			os.OpenFile(GetPathToInsecureKeyring(), os.O_CREATE, 0600)
+		}
 
-	secret, err := keyring.Get(KeyringServiceName+"."+conID, uName)
+		existingSecrets := []KeyringSecret{}
+		file, readErr := ioutil.ReadFile(GetPathToInsecureKeyring())
+		if readErr != nil {
```

Comment:
If the keyring file doesn't exist the first time you try to store a secret will this function error instead of creating the file?

Reply:
it shouldn't error, `len(file)` should be 0 and we will just write the new secret to the file. This test covers it https://github.com/eclipse/codewind-installer/pull/405/files/49e111dcb2302776e1b208b087023ef33f4ac31c#diff-5692282167dc222ced4f1485c91cbad7R94-R98